src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -373,6 +375,21 @@ public static function underscore($id)
         return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), array('\\1_\\2', '\\1_\\2'), str_replace('_', '.', $id)));
     }
 
+    protected function getEnv($name)
+    {
+        if (isset($this->envCache[$name]) || array_key_exists($name, $this->envCache)) {
+            return $this->envCache[$name];
+        }
+        if (false !== $env = getenv($name)) {
```

Comment:
Is it possible to define "providers" instead of calling `getenv` directly ?
This way, we could use key/value store (like consul, etcd, ...) or even files